lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -1818,8 +1818,31 @@ bool TypeSystemSwiftTypeRef::IsImportedType(opaque_compiler_type_t type,
   VALIDATE_AND_RETURN(impl, IsImportedType, type,
                       (ReconstructType(type), nullptr));
 }
+
 CompilerType TypeSystemSwiftTypeRef::GetErrorType() {
-  return m_swift_ast_context->GetErrorType();
+  auto impl = [&]() {
+    using namespace swift::Demangle;
+    Demangler Dem;
+    llvm::ArrayRef<NodePointer> nodes = {
+        Dem.createNode(Node::Kind::Type),
+        Dem.createNode(Node::Kind::ProtocolList),
+        Dem.createNode(Node::Kind::TypeList),
+        Dem.createNode(Node::Kind::Type),
+        Dem.createNode(Node::Kind::Protocol),
+    };
+    for (int i = 1; i < nodes.size(); ++i)
+      nodes[i - 1]->addChild(nodes[i], Dem);
+
+    auto *error_type = nodes.back();
+    auto *module =
+        Dem.createNodeWithAllocatedText(Node::Kind::Module, swift::STDLIB_NAME);
+    auto *identifier =
+        Dem.createNodeWithAllocatedText(Node::Kind::Identifier, "Error");
+    error_type->addChild(module, Dem);
+    error_type->addChild(identifier, Dem);
+    return RemangleAsType(Dem, nodes[0]);
+  };
+  VALIDATE_AND_RETURN_STATIC(impl, GetErrorType);
```

Comment:
Why not just use a string literal with the mangling? It's stable ABI.

Reply:
I discussed this with Adrian. The current answer is mostly that demangled trees are generally easier to see for what they are vs mangled symbols. In this specific case, maybe it could be a string with a comment. Are there platform portability issues to handle if using a mangled string directly?

Reply:
Yeah, I want to avoid hardcoding string literals because they are opaque, difficult to understand and maintain. My preferred implementation would be a `constexpr` demangle tree, but this is the next best thing. I suppose we could generate the string at compile time if we were concerned about performance, but I doubt that this matters.